src/nb_clean/__init__.py

```diff
@@ -180,9 +180,16 @@ def check_notebook(
             if cell["execution_count"]:
                 print(f"{prefix}: execution count")
                 is_clean = False
-            if not preserve_cell_outputs and cell["outputs"]:
-                print(f"{prefix}: outputs")
-                is_clean = False
+
+            if preserve_cell_outputs:
+                for output in cell["outputs"]:
+                    if output.get("execution_count"):
```

Comment:
Can we add an explicit `is not None`? There are non-`None` integer values for which this would evaluate to `False`: namely 0. An execution count of 0 shouldn't occur, but if it were to this could be a hard bug to track down otherwise.

Reply:
Ah, yes. We also want 0 to be changed to null. Then, it will replace null with null if we use `in`. Okay then, agreed.

Reply:
Except of we use explicit None. But in my opinion, straight replacing None with None is faster than checking if not None and replace with None.

Reply:
I will use `if "execution_count" in output:` and replace it with None then. Thanks.